client/src/main/kotlin/io/spine/examples/shareaware/client/MarketPage.kt

```diff
@@ -294,277 +322,497 @@ public class MarketPageModel(client: DesktopClient) {
+private fun PurchaseDialog(
+    model: MarketPageModel,
+    isShown: Boolean
 ) {
-    val image: T? by produceState<T?>(null) {
-        value = withContext(Dispatchers.IO) {
-            try {
-                load()
-            } catch (e: IOException) {
-                throw illegalArgumentWithCauseOf(e)
+    val scope = rememberCoroutineScope { Dispatchers.Default }
+    val shareToPurchase by model.purchaseOperation.share().collectAsState()
+    val quantity by model.purchaseOperation.quantityOfShares().collectAsState()
+    if (isShown) {
+        Dialog(
+            onCancel = {
+                scope.launch {
+                    model.purchaseOperation.cancel()
+                }
+            },
+            onConfirm = {
+                scope.launch {
+                    model.purchaseOperation.complete()
+                    model.purchaseOperation.cancel()
+                }
+            },
+            title = "Purchase '${shareToPurchase?.companyName}' shares",
+            modifier = Modifier
+                .wrapContentHeight()
+                .width(245.dp),
+            {
+                val price = multiply(shareToPurchase!!.price, quantity).asReadableString()
+                PurchaseDialogInput(
+                    price = price,
+                    model = model
+                )
             }
-        }
-    }
-    if (image != null) {
-        Image(
-            painter = painterFor(image!!),
-            contentDescription = contentDescription,
-            contentScale = contentScale,
-            modifier = modifier
         )
     }
 }
 
 /**
- * Loads an image from the network by URL.
+ * Displays the input inside the `PurchaseDialog`.
  *
- * @param url the URL of the image to load
- * @param density density that will be used to set the intrinsic size of the image
- * @return the decoded SVG image associated with the URL
- */
-private fun loadImage(url: String, density: Density): Painter =
-    URL(url).openStream().buffered().use { loadSvgPainter(it, density) }
-
-/**
- * Returns the readable `String` constructed from the `Money` object.
+ * @param price total price of the purchase to be displayed
+ * @param model the model of the "Market" page
  */
-private fun Money.asReadableString(): String {
-    return "$" + this.units.toString() + "." + this.nanos.toString()
+@Composable
+private fun PurchaseDialogInput(
+    price: String,
+    model: MarketPageModel
+) {
+    Column {
+        Text(
+            "Total Price - $price",
+            style = MaterialTheme.typography.bodySmall,
+            textAlign = TextAlign.Center,
+            modifier = Modifier.padding(start = 16.dp, bottom = 15.dp)
+        )
+        NumericInput(
+            placeholder = "How much to purchase",
+            onChange = {
+                val quantity = if (it == "") 0 else it.toInt()
+                model.purchaseOperation.quantityOfShares(quantity)
+            }
+        )
+    }
```

Comment:
As discussed previously and vocally, let's create `... .market` "package" and  split this (still) big file into several pieces.